### PR TITLE
Check whether particlePatches are dirty

### PR DESCRIPTION
**Pull request by franzpoeschel**
Fix for #907. `::dirtyRecursive()` ignored particle patches, leading to skipped flushes.
We should do a minor release soon including this one since this makes restarting from openPMD impossible in PIConGPU.

TODO:
- [x] Add a test for this

### CODE REVIEW

File: include/openPMD/backend/PatchRecordComponent.hpp
Comment (ax3l) [inline include/openPMD/backend/PatchRecordComponent.hpp:83]: Can we add one sentence here repeating what "dirty" means? :)

**Review by ax3l — COMMENTED** [review]
Looks great! Also great doxygen string.

Just waiting for the test before merge.

**Comment by psychocoderHPC** [comment]
I tested this PR together with PIConGPU and the crash during the restart is fixed.

**Review by ax3l — APPROVED** [review]

**Comment by ax3l** [comment]
Wonderful, goes in! Yes, let's cut a `0.13.2` release with this fix very soon.

**Pull request merged**
